java/src/jmri/jmrix/sprog/SprogProgrammer.java

```diff
@@ -259,8 +255,17 @@ synchronized public void notifyVersion(SprogVersion v) {
         _sv = v;
         // Save it for others
         _memo.setSprogVersion(v);
-        if (log.isDebugEnabled()) {
-            log.debug("Found: {}", v.toString());
+        log.debug("Found: {}", v.toString());
+
+        if (!_sv.supportsCVHints()) {
+            log.debug("Hardware does not support hints");
+           _startVal = 0;
+        }
+        try {
+            readCVWithDefault(_cv, _progListener, _startVal);
+        }
+        catch (jmri.ProgrammerException e) {
```

Comment:
We usually have the catch statement at the same line as the ending curling bracket.
`} catch (jmri.ProgrammerException e) {`